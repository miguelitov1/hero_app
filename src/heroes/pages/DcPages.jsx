import { HeroesList } from "../components"


export const DcPages = () => {
  return (
    <>

      <h1>DC Comics</h1>
      <hr />

      <HeroesList publisher='DC Comics'/>

    </>
  )
}
