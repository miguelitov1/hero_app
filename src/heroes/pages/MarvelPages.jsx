import { HeroesList } from "../components"


export const MarvelPages = () => {
  return (
    <>

      <h1>Marvel Comics</h1>
      <hr />

      <HeroesList publisher='Marvel Comics'/>

    </>
  )
}