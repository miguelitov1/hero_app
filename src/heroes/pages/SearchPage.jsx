import { HeroCard } from "../components/index";

export const SearchPage = () => {
  return (
    <>

      <h1>Search</h1>
      <hr />

      <div className="row">

        <div className="col-3">
          
          <h4>Searching</h4>
          <hr />

          <form>
            <input 
              type="text" 
              placeholder="Search a hero"
              className="form-control"
              name="searchText"
              autoComplete="off"
            />

            <button
              className="btn btn-outline-primary mt-1">
              Search
            </button>

          </form>

        </div>

        <div className="col-9">
          
          <h4>Results</h4>
          <hr />

          <div className="alert alert-primary">
            Search a hero
          </div>          
          
          <div className="alert alert-danger">
            There's no hero with <b>ABC</b>
          </div>

          <HeroCard />

        </div>

      </div>


    </>

  )
}
