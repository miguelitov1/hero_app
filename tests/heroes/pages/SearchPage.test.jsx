import { fireEvent, getByLabelText, render, screen } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom"
import { SearchPage } from "../../../src/heroes/pages/SearchPage"



const mockedUseNavigate = jest.fn();

//de esta forma hacemos un mock de una libreria completa!!
jest.mock( 'react-router-dom', () => ({

    ...jest.requireActual( 'react-router-dom' ),//desestructuramos para mandarle todo lo demas de la libreria que no modificamos
    useNavigate: () => mockedUseNavigate

}) )

describe('Pruebas en <SearchPage />', () => { 

    beforeEach(() => jest.clearAllMocks() );

    test('Debe de mostrarse correctamente con valores por defecto', () => { 

        const { container } = render(
            <MemoryRouter>
                <SearchPage />
            </MemoryRouter>
        )

        expect( container ).toMatchSnapshot();

     });

     test('Debe de mostrarse a Batman y el imput con el valor del queryString', () => { 

        render(
            <MemoryRouter initialEntries={[ '/search?q=batman' ]}>
                <SearchPage />
            </MemoryRouter>
        )

        const input = screen.getByRole( 'textbox' );
        expect( input.value ).toBe( 'batman' );

        const img = screen.getByRole( 'img' );
        expect( img.src ).toContain( 'assets/heroes/dc-batman.jpg' );

        const errorDiv = screen.getByLabelText( 'error_div' );
        expect( errorDiv.style.display ).toBe( 'none' );


     });

     test('Debe de mostrar un error si no se encuentra el hero ("batman123")', () => { 

        render(
            <MemoryRouter initialEntries={[ '/search?q=batman123' ]}>
                <SearchPage />
            </MemoryRouter>
        )

        const errorDiv = screen.getByLabelText( 'error_div' );
        expect( errorDiv.style.display).toContain( '' );

     });

     test('Debe de llamar al navigate a la pantalla nueva', () => { 

        const { container } = render(
            <MemoryRouter>
                <SearchPage />
            </MemoryRouter>
        )

        const input = screen.getByRole( 'textbox' );
        fireEvent.change( input, { target: {name: 'searchText', value: 'superman' } } );

        const form = screen.getByLabelText( 'form' );
        fireEvent.submit( form );

        expect( mockedUseNavigate ).toHaveBeenCalledWith( '?q=superman' );


     })

 })