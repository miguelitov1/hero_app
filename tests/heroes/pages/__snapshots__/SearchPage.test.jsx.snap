// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SearchPage /> Debe de mostrarse correctamente con valores por defecto 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-3"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form
        aria-label="form"
      >
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-1"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-9"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        class="alert alert-primary animate__animated animate__fadeIn"
      >
        Search a hero
      </div>
      <div
        aria-label="error_div"
        class="alert alert-danger animate__animated animate__fadeIn"
        style="display: none;"
      >
        There's no hero with 
        <b />
      </div>
    </div>
  </div>
</div>
`;
